### DISCUSSION TITLE
esp32/MicroPython/08.esp32-micropython-button

### DISCUSSION THREAD

**Discussion by giscus** [bigrich-luo/DocsGiscus#38] [General]
# esp32/MicroPython/08.esp32-micropython-button

https://docs.geeksman.com/esp32/MicroPython/08.esp32-micropython-button.html

**Comment by chenchangan6** [comment]
您好,我用了以下内容。但是按键接的GND，并把Pin.PULL_DOWN改为Pin.PULL_UP调试发现
按下按键ESP32主板暂停工作了。松开后继续工作。我使用的Thonny.
我测试的办法是直接print pin_button.value()的值。一直输出为1.
但是按下按钮后，并不输出0.而是程序暂停了。松开后继续输出1.

非常希望知道原因。谢谢。

引用如下：

# 创建按键输入引脚类，如果引脚的一端接 Vcc，则设置下拉电阻；如果一端接的是 GND，则配置上拉电阻。
pin_button = Pin(14, Pin.IN, Pin.PULL_DOWN)

**Reply by Bald0Wang** [reply]
你写的有点问题，如果是让Pin14做输出位应该是OUT，我是
pin_button = Pin(14, Pin.OUT, Pin.PULL_UP)
但是没有work，我觉得逻辑是通的。为啥呢？
